requirements.txt

```
@@ -1,11 +1,15 @@
 # the big split on 3.11 is required due to dwave-samplers switch, and hard
 # dependency on narrow dimod version range (also dwave-preprocessing)
 # see :pr:284
+# and then another split on 3.12 is required to install minimal versions with
+# py312 support
 
 dimod==0.10.13; python_version<"3.11"
-dimod==0.12.3; python_version>="3.11"
+dimod==0.12.3; python_version>="3.11" and python_version<"3.12"
```

Very minor, but would something this work as well? Same below.
```suggestion
dimod==0.12.3; python_version~="3.11"
```